scipy/stats/tests/test_stats.py

```diff
@@ -581,6 +581,19 @@ def test_nan_policy(self):
         assert_raises(ValueError, stats.spearmanr, x, x, nan_policy='raise')
         assert_raises(ValueError, stats.spearmanr, x, x, nan_policy='foobar')
 
+    def test_nan_policy_bug_12458(self):
+        np.random.seed(5)
+        x = np.random.rand(5, 10)
```

Comment:
```suggestion
        x = np.random.default_rng(5).uniform(0, 1, 50).reshape(5,10)
```
reason: numpy docs indicate `numpy.random.rand()` is deprecated. 

The proposal achieves the same thing. 
IIRC the deprecation of `.rand()` is (somehow) related to the switch the the `Generator` for generating random variates. If this is incorrect please let me know.

Reply:
Yeah I think `Generator` doesn't have a `rand` method.

Reply:
Hmm, maybe I was too quick on this, are we still supporting old versions of numpy that didn't have the `default_rng` method inside random? 

e.g.
https://travis-ci.org/github/scipy/scipy/jobs/726677838#L1314

looks like that's causing the build to fail. I guess we can back out this change to get the CI to pass and use the `random.rand()` approach for now.